### PR TITLE
Authorization using verified credentials; SSI integration based on Jolocom SDK

### PR DESCRIPTION
**Pull request by techsmyth**
Integration of Jolocom SDK (sqlite for now, shortly mysql)
Creation of DID identity (jun) for all agents. 
Adds Verified Credentials (VCs) support to Agent
Extended AuthorizationDefinition to be able to specify VCs
Extends AuthorizationEngine to be able to verify based on Verified Credentials.
New mutation for assigning a VC from a Challenge to User
Extended XState integration to be able to work with VC based authorization
Extended sample XState workflow definition for Challenge (default) to use a guard based on VC authorization
...

Notes:
- this is an experimental feature, that is disabled by default
- two updates needed from Jolocom before can fully leverage: support mysql via typeorm (imminent), migrations support from sdk. Latter is to be raised.

### CODE REVIEW

File: src/app.module.ts [hunk 1]
```diff
@@ -54,6 +55,34 @@ import { MembershipModule } from '@src/services/domain/membership/membership.mod
...
+      inject: [ConfigService],
+      useFactory: async (configService: ConfigService) => ({
+        type: 'sqlite', //todo: switch to mysql when issue is addressed.
+        insecureAuth: true,
+        synchronize: true /* note: only for demo */,
+        cache: true,
+        entities: [
+          'node_modules/@jolocom/sdk-storage-typeorm/js/src/entities/*.js',
+        ],
+        // host: configService.get(ConfigurationTypes.Identity)?.ssi.jolocom.database
```
Comment (valentinyanakiev) [inline src/app.module.ts:71]: why do you keep all those commented lines?

File: src/domain/agent/agent/agent.resolver.fields.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  })
+  @Profiling.api
+  async verifiedCredentials(
+    @Parent() agent: Agent
+  ): Promise<VerifiedCredential[]> {
+    const ssiEnabled = this.configService.get(ConfigurationTypes.Identity).ssi
+      .enabled;
+    if (ssiEnabled) {
+      return await this.agentService.getVerifiedCredentials(agent);
+    }
```
Comment (valentinyanakiev) [inline src/domain/agent/agent/agent.resolver.fields.ts:38]: It's better to throw an exception than returning empty array.

File: src/migrations/1623310575144-agent-auth.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class agentAuth1623310575144 implements MigrationInterface {
```
Comment (valentinyanakiev) [inline src/migrations/1623310575144-agent-auth.ts]: Do we need two migrations for one story?

**Review by valentinyanakiev — CHANGES_REQUESTED** [review]
A few minor changes are required after a quick scan - comments inline.

File: src/app.module.ts (hunk 1 above)
Comment (techsmyth) [reply src/app.module.ts:71]: am waiting for a fix from jolocom; once that is in then want to enable mysql storage.

File: src/domain/agent/agent/agent.resolver.fields.ts (hunk 2 above)
Comment (techsmyth) [reply src/domain/agent/agent/agent.resolver.fields.ts:38]: 0373ff34429161e383c18c9ee5111b08a1a8f27d

File: src/migrations/1623310575144-agent-auth.ts (hunk 3 above)
Comment (techsmyth) [reply src/migrations/1623310575144-agent-auth.ts]: 0373ff34429161e383c18c9ee5111b08a1a8f27d

**Review by valentinyanakiev — APPROVED** [review]

**Pull request merged**